aws-lc-sys/builder/cc_builder.rs

```diff
@@ -0,0 +1,297 @@
+    test_command, OutputLibType,
+};
+use std::path::PathBuf;
+
+pub(crate) struct CcBuilder {
+    manifest_dir: PathBuf,
+    out_dir: PathBuf,
+    build_prefix: Option<String>,
+    output_lib_type: OutputLibType,
+}
+
+use std::fs;
+
+pub(crate) struct Library {
+    name: &'static str,
+    flags: &'static [&'static str],
+    sources: &'static [&'static str],
+}
+
+#[allow(non_camel_case_types)]
+enum PlatformConfig {
+    aarch64_apple_darwin,
+    aarch64_unknown_linux_gnu,
+    aarch64_unknown_linux_musl,
+    x86_64_apple_darwin,
+    x86_64_unknown_linux_gnu,
+    x86_64_unknown_linux_musl,
+    i686_unknown_linux_gnu,
+}
+
+impl PlatformConfig {
+    fn libcrypto(&self) -> Library {
+        match self {
+            PlatformConfig::aarch64_apple_darwin => aarch64_apple_darwin::CRYPTO_LIBRARY,
+            PlatformConfig::aarch64_unknown_linux_gnu => aarch64_unknown_linux_gnu::CRYPTO_LIBRARY,
+            PlatformConfig::aarch64_unknown_linux_musl => {
+                aarch64_unknown_linux_musl::CRYPTO_LIBRARY
+            }
+            PlatformConfig::x86_64_apple_darwin => x86_64_apple_darwin::CRYPTO_LIBRARY,
+            PlatformConfig::x86_64_unknown_linux_gnu => x86_64_unknown_linux_gnu::CRYPTO_LIBRARY,
+            PlatformConfig::x86_64_unknown_linux_musl => x86_64_unknown_linux_musl::CRYPTO_LIBRARY,
+            PlatformConfig::i686_unknown_linux_gnu => i686_unknown_linux_gnu::CRYPTO_LIBRARY,
+        }
+    }
+
+    fn default_for(target: &str) -> Option<Self> {
+        println!("default_for Target: '{target}'");
+        match target {
+            "aarch64-apple-darwin" => Some(PlatformConfig::aarch64_apple_darwin),
+            "aarch64-unknown-linux-gnu" => Some(PlatformConfig::aarch64_unknown_linux_gnu),
+            "aarch64-unknown-linux-musl" => Some(PlatformConfig::aarch64_unknown_linux_musl),
+            "x86_64-apple-darwin" => Some(PlatformConfig::x86_64_apple_darwin),
+            "x86_64-unknown-linux-gnu" => Some(PlatformConfig::x86_64_unknown_linux_gnu),
+            "x86_64-unknown-linux-musl" => Some(PlatformConfig::x86_64_unknown_linux_musl),
+            "i686-unknown-linux-gnu" => Some(PlatformConfig::i686_unknown_linux_gnu),
+            _ => None,
+        }
+    }
+}
+
+impl Default for PlatformConfig {
+    fn default() -> Self {
+        Self::default_for(&target()).unwrap()
+    }
+}
+
+impl CcBuilder {
+    pub(crate) fn new(
+        manifest_dir: PathBuf,
+        out_dir: PathBuf,
+        build_prefix: Option<String>,
+        output_lib_type: OutputLibType,
+    ) -> Self {
+        Self {
+            manifest_dir,
+            out_dir,
+            build_prefix,
+            output_lib_type,
+        }
+    }
+
+    fn create_builder(&self) -> cc::Build {
+        let mut cc_build = cc::Build::default();
+        cc_build
+            .out_dir(&self.out_dir)
+            .flag("-std=c99")
+            .flag("-Wno-unused-parameter")
```

Comment:
why just this flag and not any of the others? do we want to keep all of these in sync with CMake? do we want to emit warnings like these in the rust build?